src/libraries/System.Linq.Queryable/src/System/Linq/EnumerableRewriter.cs

```diff
@@ -136,21 +136,9 @@ private static Type GetPublicType(Type t)
                 return typeof(IEnumerable);
             return t;
 
-            [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2075:UnrecognizedReflectionPattern",
-                Justification = "The IGrouping<,> is kept since it's directly referenced here" +
-                    "and so it will also be preserved in all places where it's implemented." +
-                    "The GetInterfaces may return less after trimming but it will include" +
-                    "the IGrouping<,> if it was there before trimming, which is enough for this" +
-                    "method to work.")]
             static bool ImplementsIGrouping(Type type) =>
```

Comment:
Ideally we would revert the code to the shape it was before https://github.com/dotnet/runtime/pull/52461/files#diff-ebbc6bf7ee73960d1921153ebc67451c142e0455a82e08a2b7c61c45cf7c2217L129

Since the refactoring was done solely to be able to target-suppress the warnings (Which is not necessary anymore, since the entire type is RUC annotated).